summarization/html_parsers/kisalfold_parser.py

```diff
@@ -0,0 +1,103 @@
+from datetime import datetime
+from typing import Optional, Set, List
+
+from bs4 import Tag
+
+from summarization.html_parsers.parser_base import ParserBase
+from summarization.utils.assertion import assert_has_article, assert_has_title
+from summarization.utils.dateparser import DateParser
+
+
+class KisalfoldParser(ParserBase):
+    def get_title(self, url, soup) -> str:
+        title = soup.find('h2', class_='single-article__title')
+
+        if not title:
+            title = soup.find('h1', class_='article-title_h1')
+
+        if not title:
+            title = soup.find('h1', id='article-title_h1')
+
+        if not title:
+            title = next(iter(soup.select('div#article_content > h2')), None)
+
+        if not title:
+            title = next(iter(soup.select('article.left-column > h1')), None)
+
+        assert_has_title(title, url)
+        return self.get_text(title)
+
+    def get_lead(self, soup) -> Optional[str]:
+        lead = soup.find('div', class_='single-article__lead')
+
+        if not lead:
+            lead = soup.find('div', class_='enews-article-lead')
+
+        if not lead:
+            lead = soup.find('h4', id='article_lead')
+
+        if not lead:
+            lead = next(iter(soup.select('article.left-column > p.lead')), None)
+
+        return self.get_text(lead, '')
+
+    def get_article_text(self, url, soup) -> str:
+        article_text = self.get_text(soup.find('div', id='article_text'))
+
+        if not article_text:
+            article = soup.select('div.enews-article-content > p')
```

Comment:
Are you sure the p tag is enough? What if the article has a list too? (also the same problem below this)

Reply:
Should I generalize and just use `div.enews-article-content`? Every article I've looked at had 1 or 2 `p` tags, but there could be a case when something else is present.

Reply:
Yes, I think that should be enough

Reply:
Found an example that gets parsed wrong:
https://www.kisalfold.hu/orszag-vilag/2014/08/egy-heten-belul-johetnek-az-ujabb-oroszellenes-szankciok